ros2/cc_defs.bzl

```diff
@@ -64,7 +64,39 @@ def ros2_c_binary(name, ros2_package_name = None, **kwargs):
     """
     _ros2_cc_target(cc_binary, "c", name, ros2_package_name, **kwargs)
 
-def ros2_cpp_binary(name, ros2_package_name = None, **kwargs):
+def _ros2_cpp_exec(target, name, ros2_package_name = None, set_up_ament = False, **kwargs):
+    if set_up_ament == False:
+        _ros2_cc_target(target, "cpp", name, ros2_package_name, **kwargs)
+        return
+
+    target_impl = name + "_impl"
+    _ros2_cc_target(cc_binary, "cpp", target_impl, ros2_package_name, tags = ["manual"], **kwargs)
+
+    is_test = target == cc_test
+
+    launcher = "{}_launch".format(name)
+    sh_launcher(
+        launcher,
+        deps = [target_impl],
+        template = "@com_github_mvukov_rules_ros2//ros2:launch.sh.tpl",
+        substitutions = {
+            "{entry_point}": "$(rootpath {})".format(target_impl),
+        },
+        tags = ["manual"],
+        data = [target_impl],
+        testonly = is_test,
+    )
+
+    sh_target = native.sh_test if is_test else native.sh_binary
+    sh_target(
+        name = name,
+        srcs = [launcher],
+        data = [target_impl],
+        tags = kwargs.pop("tags", []),
+        visibility = kwargs.pop("visibility", None),
```

Comment:
Fair enough (at least for now).

Reply:
I agree. It's done locally, but since GH is experiencing issues currently, I can't push right now. Will ping you once you can have a (hopefully) final look.